codec/helper.go

```diff
@@ -1564,41 +1564,42 @@ func isEmptyStruct(v reflect.Value, tinfos *TypeInfos, deref, checkStruct bool)
 // 	return t
 // }
 
-func panicToErr(h errstrDecorator, err *error) {
+func panicToErr(h errDecorator, err *error) {
 	// Note: This method MUST be called directly from defer i.e. defer panicToErr ...
 	// else it seems the recover is not fully handled
 	if recoverPanicToErr {
 		if x := recover(); x != nil {
 			// fmt.Printf("panic'ing with: %v\n", x)
 			// debug.PrintStack()
-			panicValToErr(h, x, err)
+			*err = panicValToErr(h, x)
 		}
 	}
 }
 
-func panicValToErr(h errstrDecorator, v interface{}, err *error) {
+func panicValToErr(h errDecorator, v interface{}) error {
```

Comment:
Making this and errDecorator.wrapErr return errors instead of filling one in seemed cleaner to me. But maybe there's a good reason for the old way?

Reply:
With the old way, the err parameter is only updated if it should be. 
If the old err was non-nil, then a call to panicValToErr would cause it to stay non-nil even if the v is nil.

In your current way, the err parameter is always updated i.e. 
if the old err was non-nil, then a call to panicValToErr can cause it to become nil if the v is nil.

Reply:
Ah, ok. I guess that matters for updating e.err and d.err?